File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Shared/ResponseJsons/UserUpdateResponse.cs
using BeltmanSoftwareDesign.Shared.Jsons;

namespace BeltmanSoftwareDesign.Shared.ResponseJsons
{
    public class UserUpdateResponse : Response
    {
        public User? User {  get; set; }
        public bool ErrorOnlyUpdatesToYourselfAreAllowed { get; set; }
    }
}

File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Angular/src/app/interfaces/response/userupdateresponse.ts
import { User } from "../user";
import { State } from "../state";

export interface UserUpdateResponse {
    user: User | null;
    errorOnlyUpdatesToYourselfAreAllowed: boolean;
    success: boolean;
    errorAuthentication: boolean;
    errorItemNotFound: boolean;
    state: State | null;
}